scipy/ndimage/tests/test_morphology.py

```diff
@@ -441,6 +441,26 @@ def test_distance_transform_cdt04(self):
                 indices=indices_out
             )
 
+    @pytest.mark.parametrize('dtype', types)
+    def test_distance_transform_cdt05(self, dtype):
+        # test custom metric type per dicussion on issue #17381
+        data = numpy.array([[0, 0, 0, 0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 1, 1, 1, 0, 0, 0],
+                            [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                            [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                            [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                            [0, 0, 0, 1, 1, 1, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype)
+        metric_arg = numpy.array([[1, 1, 1],
+                                  [1, 1, 1],
+                                  [1, 1, 1]])
+        try:
+            ndimage.distance_transform_cdt(data, metric=metric_arg)
+        except Exception as exc:
+            raise pytest.fail("Unit test failed. Raised {0}".format(exc))
```

Comment:
The test should simply check some known result. The try block is useless since if it would fail, the test would fail automatically.

Reply:
I've tried to revise it accordingly.